src/AzureAppConfigurationImpl.ts

```diff
@@ -479,17 +336,141 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
     }
 
     /**
-     * Refresh key-values.
+     * Loads configuration settings from App Configuration, either key-value settings or feature flag settings.
+     * Additionally, updates the `pageEtags` property of the corresponding @see PagedSettingSelector after loading.
+     *
+     * @param loadFeatureFlag - Determines which type of configurationsettings to load:
+     *                          If true, loads feature flag using the feature flag selectors;
+     *                          If false, loads key-value using the key-value selectors. Defaults to false.
+     */
+    async #loadConfigurationSettings(loadFeatureFlag: boolean = false): Promise<ConfigurationSetting[]> {

```

Comment:
nit: You could rewrite this function to get rid of the boolean flag. I think it's a little cleaner and more generic. It would look like:

```typescript
    async #loadConfigurationSettings(selectors: PagedSettingSelector[]): Promise<ConfigurationSetting[]> {
        const funcToExecute = async (client) => {
        ...
        }
        
        // Override selectors array with selectors to update array.
        selectors.splice(0, selectors.length, ...selectorsToUpdate);
        
        ...
```

Reply:
I think we have to keep the boolean flag. The reason is that the selector can just be keyFilter="*" which will load all key-values includes feature flags. However, when featureFlagOption is not enabled, we should not load feature flag to the configuration. That's why I have these lines:

``` js
for (const setting of page.items) {
   if (loadFeatureFlag === isFeatureFlag(setting)) {
      loadedSettings.push(setting);
  }
}
```